db/schema.rb

```diff
@@ -0,0 +1,64 @@
+# This file is auto-generated from the current state of the database. Instead
+# of editing this file, please use the migrations feature of Active Record to
+# incrementally modify your database, and then regenerate this schema definition.
+#
+# This file is the source Rails uses to define your schema when running `bin/rails
+# db:schema:load`. When creating a new database, `bin/rails db:schema:load` tends to
+# be faster and is potentially less error prone than running all of your
+# migrations from scratch. Old migrations may fail to apply correctly if those
+# migrations use external dependencies or application code.
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema[7.0].define(version: 2023_03_10_202314) do
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "comments", force: :cascade do |t|
+    t.string "text"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "user_id", null: false
+    t.bigint "post_id", null: false
+    t.index ["post_id"], name: "index_comments_on_post_id"
+    t.index ["user_id"], name: "index_comments_on_user_id"
+  end
+
+  create_table "likes", force: :cascade do |t|
+    t.integer "authorid"
+    t.integer "postid"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "user_id", null: false
+    t.bigint "post_id", null: false
+    t.index ["post_id"], name: "index_likes_on_post_id"
+    t.index ["user_id"], name: "index_likes_on_user_id"
+  end
+
+  create_table "posts", force: :cascade do |t|
+    t.integer "authorid"
+    t.string "title"
+    t.string "text"
+    t.integer "commentscounter"
+    t.integer "likescounter"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "user_id", null: false
```

Comment:
- [ ] If you take a look at your `schema.rb` file, you will realize that you have both `authorid` and `user_id` columns on the `posts`, and `likes` tables.
This is because in your `create_posts` and `create_likes` migration files, you had specified the `authorid` column, and still went ahead to create a separate migration file to add `user` reference to both the `posts` and `likes` table.

The easiest solution would be for you to have this in your create_posts, create_likes and create_comments migration files
```ruby
t.references :author, foreign_key: { to_table: 'users' }
```
In that case, there'd be no need for you to have `t.integer :authorid` in the migration anymore because the command I showed you above will create `author_id` and a corresponding` index`